### ISSUE TITLE
Track/reminder of uninstall notes

### ISSUE THREAD

**Issue by Shine753** [ModOrganizer2/modorganizer#1088]
Being able to take a note when uninstalling a mod, and keep track of it in a separate interface.
And if trying to reinstall the same mod, display the user this reason of why he uninstalled it the last time.
And of course, being able to delete this uninstall note to start again with a clean situation.

**Comment by isanae** [comment]
Uninstalling a mod deletes its directory completely, including the `meta.ini` file where we store per-mod information. We talked about storing some information in other places, which would allow for this sort of feature, but don't hold your breath.

This is also oddly specific.

**Comment by Al12rs** [comment]
Yeah I think this is too specific and beyond the scope of Mo2, with limited use cases and limited possibility to streamline this into the current workflow. It don't want to overcrowd the workflow with steps that aren't needed.
I can reconsider if enough reason is given and there is a good proposition on how to add this in a non intrusive way that doesn't make the feature itself unusable.

**Comment by Shine753** [comment]
Ok, I'll think about this and try to argument some solutions here.

**Comment by Shine753** [comment]
To investigate the way you think this could be, or not, possible.
Are you opened to this kind of idea :
Having a dedicated "uninstalled" folder into the MO2 mods folder. When you uninstall a mod, it moves the meta.ini in this folder, and check them when you install a mod. This way, the meta.ini could contain the uninstall reason, and all the other metas. So when reinstalling a old uninstalled mod, If it finds the old meta.ini, you would not loose all the metas you had already put before on the mod.
And if needed, just a menu entry to purge this folder could cleanup the whole too old entries.

**Comment by Al12rs** [comment]
What I'm more concerned about is how Mo2 is supposed to ask the user why they wanted to delete the mod. I feel this would just be interrupting your workflow however you do it. So this could be an idea for a Mo2 plugin (we support python if you want to try yourself), but I don't see this as being part of Mo2 proper.

**Comment by Shine753** [comment]
I agree, that sounds good for a plugin instead of a native behaviour.
But I won't be able to handle this, I don't know Python.

**Comment by Al12rs** [comment]
If you know any other object oriented programming language you can learn the python you need in a couple of hours or less.

**Comment by Shine753** [comment]
For my information, MO2 seems to be developed in C++. Then why having chosen Python for the plugins ?

**Comment by isanae** [comment]
MO only supports C++ plugins. However, we have a C++ plugin that adds support for Python plugins, mostly for convenience. Some people like Python.

**Comment by AnyOldName3** [comment]
It's not any less built-in than support for any of the games or installers or a myriad of other features, though, as nearly everything's implemented as a plugin, so don't get the impression that it's tacked on as a gimmick. The Python interface is *supposed* to work, and these days, usually does.
